### PR TITLE
python{39,310,311,312,313}: Allow using cc to resolve dynamic libraries

### PR DESCRIPTION
**Pull request by rhelmot**
## Description of changes

This patch hasn't been complete since [an alternate case was added to cpython 3.6](https://github.com/python/cpython/commit/82df3b3071bb003247c33eac4670775e9883c994) which allows using ld(1) to resolve libraries. In addition to this, the stated reason for the patch, to improve startup times by preventing cpython from invoking gcc to resolve libuuid, has not been an issue since [that logic was removed in cpython 3.9](https://github.com/python/cpython/commit/d6b727e2c947240804b8e434b305ba2890122550). Finally, this patch [creates a disparity between Linux and other operating systems](https://github.com/NixOS/nixpkgs/pull/328776) (I am working on FreeBSD right now) since the ld(1) case is system gated.

Since it no longer accomplishes its stated purpose, is no longer needed, and creates platform disparities, we should remove it.

I've left the half of this patch which prevents /sbin/ldconfig from being invoked, since no nix-compiled program should ever be invoking absolute paths like this.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]
Makes sense to me, but perhaps a Python maintainer should review it

**Review by fpletz — APPROVED** [review]

**Comment by natsukium** [comment]
The change looks fine to me.
I'm running `python{39,310,311,312,313}.tests` on x86_64-linux, aarch64-linux, x86_64-darwin and aarch64-darwin.

**Review by natsukium — APPROVED** [review]
All tests passed. Thanks.

**Pull request merged**
